include/abstract_data_store.h

```diff
@@ -18,6 +18,8 @@ template <typename data_t> class AbstractDataStore
   public:
     AbstractDataStore(const location_t capacity, const size_t dim);
 
+    virtual ~AbstractDataStore() = default;
```

Comment:
this was an unrelated issue that address sanitizer picked up. Index holds a pointer of type AbstractDataStore but the destructor in that class is not virtual, so the InMemDataStore destructor is not invoked when the data store is freed